src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -459,11 +477,29 @@ public virtual void WriteDeltaDeletedEntry(object graph, ODataWriter writer, ODa
         /// <param name="writeContext">The <see cref="ODataSerializerContext"/>.</param>
         public virtual async Task WriteDeltaDeletedEntryAsync(object graph, ODataWriter writer, ODataSerializerContext writeContext)
         {
-            ODataDeletedResource deletedResource = GetDeletedResource(graph);
-            if (deletedResource != null)
+            ODataResourceSerializer serializer = SerializerProvider.GetEdmTypeSerializer(_elementType) as ODataResourceSerializer;
+
+            if (serializer != null)
+            {
+                ResourceContext resourceContext = serializer.GetResourceContext(graph, writeContext);
+                SelectExpandNode selectExpandNode = serializer.CreateSelectExpandNode(resourceContext);
+
+                if (selectExpandNode != null)
+                {
+                    ODataDeletedResource deletedResource = GetDeletedResource(graph, resourceContext, serializer, selectExpandNode, writeContext.IsUntyped);
+
+                    if (deletedResource != null)
```

Comment:
Based on the implementation of `GetDeletedResource` method, can `deletedResource` be null really or is this check superfluous?

Reply:
Yea It could be superflous